src/plans/solver_state.jl

```diff
@@ -242,6 +242,21 @@ For the point storage use `PointStorageKey`. For tangent vector storage use
 Initialize the Functor to an (empty) set of keys, where `once` determines
 whether more that one update per iteration are effective
 
+    function StoreStateAction(
+        M::AbstractManifold,
+        dictionary_symbols::Vector{Symbol},
+        ::Type{TPS},
+        ::Type{TTS};
+        p_init=rand(M),
+        X_init=zero_vector(M, p_init),
+        once=true,
+    ) where {TPS<:Tuple,TTS<:Tuple}
+
+Initialize the general storage keys to `dictionary_symbols`, point storage keys to `TPS` and
+tangent vector storage tu `TTS`. For example you may call
+`StorageStateAction(M, Symbol[], Tuple{:Iterate}, Tuple{:Gradient})` to create efficient
```

Comment:
Ah this is good to read here, because just from `Type{TPS}`  I would have thought you pass the point type in here. Why does it have to be the type of a tuple and not a tuple? Just curious, because maybe a tuple might be even easier to use here – but sure this is a good improvement over the more-internal other constructor. Thanks!

Reply:
I actually like the new constructor too :slightly_smiling_face: .

It's `Tuple{:Iterate}` instead of `(:Iterate,)` basically just because I don't trust Julia's constant propagation enough to handle the symbols right, and using a tuple type forces Julia to propagate these keys.